protocol/carbon/carbon_test.go

```diff
@@ -30,13 +34,16 @@ func TestNewCarbonDatapoint(t *testing.T) {
 	_, err = NewCarbonDatapoint("hello bob 3", identityParser)
 	assert.NotEqual(t, nil, err, "Line should be invalid")
 
+	floatDp, _ = NewCarbonDatapoint("hello 3.3 1519398226.544148", identityParser)
+	assert.Equal(t, int64(1519398226544), floatDp.Timestamp.UnixNano()/1000000, "Should get floating timestamp back as an int64")
```

Comment:
my personal preference is to equate directly with the value, not munge it to fit. Dividing by 1M will do some rounding of it's own. Please check the exact int64 that the floatDp.Timestamp is.